GC Fixes for Upstream Update

bholmes opened the pull request:
Need to implement a simple version of Frozen Segments

Adding "FEATURE_UNITY_NULLGC_WRITEBARRIER_PATCH"
to workaround write barrier problems with our GC. We need to
remove FEATURE_UNITY_NULLGC_WRITEBARRIER_PATCH
completely  when we move back to the default GC.

am11 committed:
Remove calling convention modifiers from cpuid sig (#83672)

> /runtime/src/coreclr/vm/cgensys.h:107:26: error: stdcall calling convention is not supported on builtin function [-Werror,-Wignored-attributes]

cston committed:
SyntaxValueProvider: avoid performance issue with syntax list containing many items (#83483)

EgorBo committed:
Improve RA for LowerBlockStore (#83627)

kunalspathak committed:
Use BitOperations::PopCount() in genCountBits() (#83661)

tannergooding committed:
Cleanup some HWIntrinsic logic to ensure the right gtType and simdSize are being set (#83516)

* Cleanup some HWIntrinsic logic to ensure the right gtType and simdSize are being set

* Apply formatting patch

* Add a missing GetLower call

* Fix an assert for lowering TYP_SIMD12 where its handled as TYP_SIMD16

* Ensure GetLower is used in Dot for TYP_SIMD32

* Apply formatting patch

* Insert after, not before, for the _GetLower to avoid a codegen regression

* Put the _GetLower in the right place to avoid the codegen regression

* Don't change the simd size of TYP_SIMD8/12 DotProduct unnecessarily

IDisposable committed:
Correct doc comment for IMultiplyOperators returns (#83693)

Replaced _divided-by_ with _multiplied-by_ in both
IMultiplyOperators.cs and NFloat.cs.
Fixes https://github.com/dotnet/runtime/issues/80521

dotnet-bot committed:
Localized file check-in by OneLocBuild Task: Build definition ID 679: Build ID 2140030 (#83703)

jandupej committed:
[mono][jit] Adding compare all/any intrinsics. (#83515)

* xxAny, xAll comparisons in progress.

* xxAny, xxAll comparisons, part 2.

* [mono][jit] Adding compare all/any operations. Fixed umov,smov macros.

* Removed superfluous changes.

* Restored newline at the end of HelloWorld.

* Fixed unmatched brace.

* Indentation.

* Normalized boolean values to 0/1. SIMD_EXTR_ constants have friendlier names. Equality/Inequality are now also intrinsics.

* Fixed element type for comparisons.

* Temporarily disabled intrinsics. Will be permanenty reenabled once all are implemented.

SwapnilGaikwad committed:
Replace a load with cheaper mov instruction when possible (#83458)

steveisok committed:
[tvOS] Bump to new OSX 13 AppleTV queue (#83272)

jandupej committed:
[mono][jit] Add vector horizontal sums and ToScalar on arm64. (#83675)

* SN_Sum operation on arm64. Fixed dup. Replaced addv, addp, faddp with their generalized variants. Added OP_EXTRACTx opcodes to arm64 codegen. Added horizontal sums.

* Fixed smov macro. Added SN_ToScalar. Fixed code style.

* Fixed vector sums of nint/nuint.

* Temporarily disable intrinsics, until all are implemented.

ilonatommy committed:
[browser] Fix encoding problem when publishing with AOT (#83510)

* Fix.

* Regex approach is not necessary.

* Fix encoding on Windows.

* Fixed build error.

* Reverted unnecessary changes. Blocked relink with unicode.

* Revet + nit.

* Applied @kg's review.

kotlarmilos committed:
[mono][aot] Load AOT module of a container assembly using assembly name (#83711)

* Load AOT module of a container assembly using assembly name

* Use mono_image_init to init the image

* Implement mono_loader_lock on load_container_amodule

* Avoid recursive invocation by setting container_assm_name to NULL

bholmes committed:
Revert "Formatting license files to match Unity legal standardization (#45)"

This reverts commit 8c08cd3e4e357aff66e6222322c5a70165a1273b.

bholmes committed:
Update to net8.0 in the Unity files

bholmes committed:
Get Frozen Segment working with the Unity NULL GC

bholmes committed:
Unity GC write barrier patch

This is a temporary patch to allow our NULL GC to work with upstream
changes.  Check the upper bounds in JIT_WriteBarrier_PreGrow64 and
JIT_WriteBarrier_WriteWatch_PreGrow64.

We must remove this when we return to default coreclr gc.

bholmes committed:
Merge from MSFT main

Merge commit 'edb161ab06ddc69d27aee2c0e990a60221ebbe92' into unity-upstream-tracking/staging

bholmes committed:
Fixing Unity tests for Linux

Casing on the platform name changed

joncham commented on src/coreclr/vm/amd64/jithelpers_fastwritebarriers.S:
```diff
@@ -32,6 +32,23 @@ PATCH_LABEL JIT_WriteBarrier_PreGrow64_Patch_Label_Lower
+        cmp     rsi, r8
+
+#ifdef FEATURE_MANUALLY_MANAGED_CARD_BUNDLES
+        .byte 0x73, 0x4b
```
Is this a jump instruction?

bholmes commented on src/coreclr/vm/amd64/jithelpers_fastwritebarriers.S:
```diff
@@ -415,6 +432,24 @@ PATCH_LABEL JIT_WriteBarrier_WriteWatch_PreGrow64_Patch_Label_Lower
+        cmp     rsi, r10
+
+#ifdef FEATURE_MANUALLY_MANAGED_CARD_BUNDLES
+        .byte 0x73, 0x3b
```
The offset of 0x3b is probably wrong as there is an extra shr instruction on line 468.

Not sure what I am going to do about it as it may not be used.

bholmes replied:
I think I have a better way to do this...

TLDR;
We need to force the bReqUpperBoundsCheck bool to be true here...

https://github.com/dotnet/runtime/blob/main/src/coreclr/vm/amd64/jitinterfaceamd64.cpp#L833

Will swap this out and try again.

bholmes commented:
What I think is going on with the tests...
 - Windows x86 is broken upstream
 - OSX amd64 May be related to my Write barrier patch, will try something else
 - OSX arm64 (not run on the server) Failing, probably due to WB code that needs attention for arm

bholmes committed:
Force write barrier code to use the upper bounds

With the null gc, we always want to have the upper bounds check
This ensures that the tiny bounds we setup will always miss.

Remove this when we remove the null GC.

bholmes committed:
Revert "Unity GC write barrier patch"

This reverts commit 5522dd8d606a7f40e1908e24d68eb95eb4e0d4a1.
